text-encode-transform.js

```diff
@@ -25,52 +25,58 @@
     throw new ReferenceError('TextDecoder implementation required');
   }
 
-  if ('readable' in self.TextEncoder.prototype &&
-      'writable' in self.TextEncoder.prototype &&
-      'readable' in self.TextDecoder.prototype &&
-      'writable' in self.TextDecoder.prototype) {
-    return;
-  }
-
+  // These symbols end up being different for every realm, so mixing objects
+  // created in one realm with methods created in another fails.
+  const codec = Symbol('codec');
   const transform = Symbol('transform');
 
-  function addReadableAndWritable(prototype, constructor) {
-    function readable() {
-      if (!this[transform]) {
-        this[transform] = constructor(this);
-      }
-      return this[transform].readable;
+  class TextEncoderStream {
+    constructor() {
+      this[codec] = new TextEncoder();
+      this[transform] =
+          new TransformStream(new TextEncodeTransformer(this[codec]));
     }
+  }
 
-    function writable() {
-      if (!this[transform]) {
-        this[transform] = constructor(this);
-      }
-      return this[transform].writable;
+  class TextDecoderStream {
+    constructor(label = undefined, options = undefined) {
```

Comment:
Just curious, why do we need to have "= undefined"? When omitted these parameters are initialized with undefined, right?

Reply:
Yes, but then TextDecoderStream.length would be 2. By providing default arguments, we can make it be 0 so that the idlharness.html tests pass.

Reply:
Thank you!